gcs_light_client.mediawiki

```diff
@@ -0,0 +1,1056 @@
+<code>P</code> (<code>1 << false_positive_rate</code>) are required by the light
+client in order to properly incrementally decode, query, and validate
+(reconstruct from Bitcoin block) a compact filter. The parameter <code>N</code>
+cannot be known ahead of time, therefore we define the serialization of a
+compact filter of type <code>0x00</code> and <code>0x01</code> as:
+<pre>
+N || raw_filter_bytes
+</pre>
+where <code>N</code> is serialized as a 32-bit big-endian integer.
+
+However, there exists a special case of a <code>null</code> filter. This this
+case an empty byte slice is transmitted rather than consuming
+<code>4-bytes</code> to encode the size of zero.
+
+However, as the parameter <code>P</code> ''MUST'' be globally agreed upon (for
+a particular filter type), we define this value ''statically'' for filter
+types: <code>0x00</code> and <code>0x01</code>. For the two aforementioned
+filter types, the false positive rate MUST be: <code>20</code>, meaning the
+parameter <code>P</code> is: <code>2^20</code>, meaning <code>fp=20</code>.
+This value was chosen as during simulations it was the value that minimized the
+bandwidth utilized by the expected number of blocks downloaded due to false
+positives, and the bandwidth used to download the filters themselves. The code
+along with a demo used for the parameter tuning can be found [here https://github.com/Roasbeef/bips/blob/83b83c78e189be898573e0bfe936dd0c9b99ecb9/gcs_light_client/gentestvectors.go]
+
+
+=== Protocol Version Bump ===
+
+As this BIP defines new peer-to-peer behavior, we bump the protocol version by
+one in order to distinguish the newly defined behavior. Full-nodes implementing
+this BIP should advertise a protocol version of: <code>70016</code>.
+
+== New Wallet Capabilities Enabled ==
+
+The new light client mode enables wallet to maintain a very compact client-side
+index of (possibly) the entire chain. Such an index provides a great degree of
+utility for wallets, as they're now able to perform tasks such as private key
+imports and full HD-seed imports without the need of a trusted third-party
+server. Additionally, the compact client-side chain index also opens up the
+door to smart contract applications which require agent action in response to
+on-chain events. Examples of such applications include Lightning.
+
+== Backwards Compatability ==
+
+This light client protocl is NOT backwards compatible with BIP 37. Full nodes
+MAY implement both protocols to serve both types of light clients.
+
+
+== Implementation Notes ==
+
+This filter header chain SHOULD be utilized by light clients to gain a greater
+degree of security against bamboozling full-nodes during their initial chain
+sync. In addition to fetching all the bitcoin headers, light clients
+implementing this BIP SHOULD also fetch all the ''filter headers'' from
+''each'' of their connected peers. With these headers, light clients SHOULD
```

Comment:
I'd say they SHOULD only fetch the top header after doing a headers sync from all of their peers except one, from which they build the whole header tree.